### PR TITLE
Use name resolver 2.0 for module descendance checks

### PR DESCRIPTION
**Pull request by powerboat9**

### CODE REVIEW

File: gcc/rust/resolve/rust-forever-stack.h [hunk 1]
```diff
@@ -521,6 +521,12 @@ template <Namespace N> class ForeverStack
 
   std::string as_debug_string ();
 
+  /**
+   * Used to check if a module is a decendent of another module
+   * Intended for use in the privacy checker
+   */
+  bool is_module_decendent (NodeId parent, NodeId child) const;
```
Comment (CohenArthur) [inline gcc/rust/resolve/rust-forever-stack.h:528]: typo
```suggestion
  bool is_module_descendant (NodeId parent, NodeId child) const;
```

but also, I'd be happy with something like `is_module_child` which seems a bit simpler

**Review by CohenArthur — APPROVED** [review]

File: gcc/rust/resolve/rust-forever-stack.h (hunk 1 above)
Comment (powerboat9) [reply gcc/rust/resolve/rust-forever-stack.h:528]: I thought I had it spelled right, but it looks like I [didn't](https://www.merriam-webster.com/dictionary/decedent)

Comment (powerboat9) [reply gcc/rust/resolve/rust-forever-stack.h]: Should be fixed now

**Review by CohenArthur — APPROVED** [review]
thanks!

**Pull request merged**